fenix/app/src/main/java/org/mozilla/fenix/ext/FragmentActivity.kt

```diff
@@ -0,0 +1,181 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.ext
+
+import androidx.annotation.IdRes
+import androidx.fragment.app.FragmentActivity
+import androidx.navigation.NavController
+import mozilla.components.browser.state.action.ContentAction
+import mozilla.components.browser.state.search.SearchEngine
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.storage.HistoryMetadataKey
+import mozilla.components.support.ktx.kotlin.isUrl
+import mozilla.components.support.ktx.kotlin.toNormalizedUrl
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.R
+import org.mozilla.fenix.browser.browsingmode.BrowsingMode
+
+/**
+ * Navigates to the browser fragment and loads a URL or performs a search (depending on the
+ * value of [searchTermOrURL]).
+ *
+ * @param navController The [NavController] to use for navigation.
```

Comment:
`navController` & `browsingMode` are now a param that are passed in by the caller

Reply:
If my below suggestions are taken to remove the `navController` parameter (unless it is necessary for a case I didn't see) and to update `browsingMode` to use a default `null` parameter, then we can revert many of the changes to callsites and shrink the diff substantially.